Automattic/wp-calypso
# Allow for serving development fallback builds

`npm run start` is a convenient way of getting a development build of Calypso that watches changes and builds on the fly as they happen. Since most development work takes place in modern browsers, the build target for `npm run start` is 'evergreen'; that is, our set of supported evergreen browsers.

However, it's sometimes useful to be able to spin up a development server targeting older browsers, to e.g. track down a bug. This PR makes it easy to do that, by running `npm run start-fallback` or alternatively `DEV_TARGET=fallback npm run start`.

Note that this PR focuses on `npm run server`, and doesn't add support for `DEV_TARGET` to any other tasks.

#### Changes proposed in this Pull Request

* Add new `DEV_TARGET` environment variable, to specify which target to build for, when in development. When missing, `'evergreen'` is assumed.
* Update build files and server code to handle new environment variable.

#### Testing instructions

To test fallback build:

* `npm run clean`
* `npm run start-fallback`
* Open `calypso.localhost:3000` and ensure that JS files are coming from `/calypso/fallback/`. Open one of them and ensure that there is no ES2015+ syntax. Try searching for e.g. `const ` (don't forget the space), and ensure that there are no instances of it outside of comments.

To test that the default is still an evergreen build:

* `npm run clean`
* `npm run start`
* Open `calypso.localhost:3000` and ensure that JS files are coming from `/calypso/evergreen/`. Open one of them and ensure that there is some ES2015+ syntax. Try searching for e.g. `const ` (don't forget the space), and ensure that there are some instances of it outside of comments.
